adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -135,9 +146,15 @@ public void run() {
                             authenticationRequest, mTokenCacheAccessor);
                     final AuthenticationResult authResult
                             = acquireTokenSilentHandler.acquireTokenWithRefreshToken(refreshToken);
+                    mAPIEvent.setWasApiCallSuccessful(true);
                     callbackHandle.onSuccess(authResult);
                 } catch (final AuthenticationException authenticationException) {
+                    mAPIEvent.setWasApiCallSuccessful(false);
                     callbackHandle.onError(authenticationException);
+                } finally {
+                    Telemetry.getInstance().stopEvent(authenticationRequest.getTelemetryRequestId(), mAPIEvent,
+                            mAPIEvent.getEventName());
+                    Telemetry.getInstance().flush(authenticationRequest.getTelemetryRequestId());
```

Comment:
>elemetry.getInstance().flush(authenticationRequest.getTelemetryRequestId()); [](start = 21, length = 76)

Don't know if it's possible, but can you put the flush inside the stopEvent itself, you can check if it's API event, that way I think the code is more reliable, less error prone.